src/config/ast.rs

```diff
@@ -89,14 +88,16 @@ impl MatchExpr {
             _ => None,
         };
         for case in &self.cases {
-            if (cfg!(unix) && case.0.exprtype == ExprType::Unix)
-                || os.as_ref().map(|x| *x == case.0.exprtype).unwrap_or(false)
+            if (cfg!(unix) && case.0.exprtype == ExprType::Unix) // is unix
+                || os.as_ref().map(|x| *x == case.0.exprtype).unwrap_or(false) // OS matches the expr
+                || case.0.exprtype == ExprType::Any
+            // this expr is the default one
             {
                 // it matches
-                return &case.1;
+                return Some(&case.1);
```

Comment:
References to what `case.0` and `case.1` represent are unclear. Perhaps storing it in separate named variables at the start of the function would help:

```rust
let expr = case.0;
let spec = case.1;
```

Other solutions to improve readability in this regard should be taken into account as well.